### DISCUSSION TITLE
Send email after saving xyzRow entity

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6871] [Help]
Hi all,

I've overridden the AfterSave() method in xyzRowSaveHandler.cs file and I want to send an email.

How I can access EmailSender in this method or in save handler file?

Thank you,
Elena.

**Comment by edwardch** [comment]
Check out this discussion https://github.com/orgs/serenity-is/discussions/6349 

You may have some luck trying something like that out...

**Reply by elenaaralla** [reply]
Hi @edwardch,

thank you.

I've simply used injection, i.e.:

```

public class xyzSaveHandler : SaveRequestHandler<MyRow, MyRequest, MyResponse>, IxyzSaveHandler
{
    IEmailSender _emailSender;

    public xyzSaveHandler(IRequestContext context, IEmailSender emailSender)
            : base(context)
    {
        _emailSender = emailSender;
    }
...
}
```

and _emailSender is initialized and ready to use :)

have a nice day,
Elena.
